extensions/wikia/Recirculation/services/CakeRelatedContentService.php

```diff
@@ -153,13 +153,13 @@ private function formatTitle( Content $content ) {
 		return $content->getTitle();
 	}
 
-	private function getAuthor( $metadata ) {
-		if ( array_key_exists( "authorName", $metadata ) ) {
-			return $metadata["authorName"];
-		} else {
+	private function getAuthor( Content $content ) {
+		$authorObjects = $content->getAuthors();
+		if ( !is_array( $authorObjects ) || empty( $authorObjects ) ) {
 			return "";
 		}
 
+		return $authorObjects[0]->getUsername();
 	}
 
 	private function getRecirculationContentType( $contentType ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)